core/preloop.gms

```diff
@@ -137,6 +137,16 @@ p_agriEmiPhaseOut(t)$(t.val ge 2040) = 1;
 *** Rescale non-co2 base line emissions from agriculture 
 pm_macBaseMagpie(t,regi,enty)$(emiMac2sector(enty,"agriculture","process","ch4") OR emiMac2sector(enty,"agriculture","process","n2o"))
   = (1-p_agriEmiPhaseOut(t)*c_BaselineAgriEmiRed)*pm_macBaseMagpie(t,regi,enty);
+  
+*** The N2O emissions generated during biomass production in agriculture (in MAgPIE)
+*** are represented in REMIND by applying the n2obio emission factor (zero in coupled runs)
+*** in q_macBase. In standaolne runs the resulting emissions need to be subtracted (see below)
+*** from the exogenous emission baseline read from MAgPIE, since the baseline already implicitly 
+*** includes the N2O emissions from biomass. In q_macBase in core/equations.gms the N2O 
+*** emissions resulting from the actual biomass demand in REMIND are then added again. 
+
+pm_macBaseMagpie(t,regi,"n2ofertin") = pm_macBaseMagpie(t,regi,"n2ofertin") - (p_efFossilFuelExtr(regi,"pebiolc","n2obio") * pm_pebiolc_demandmag(t,regi));
```

Comment:
I am not sure I fully understand this (always only talking about the REMIND standalone version)

Is pm_pebiolc_demandmag updated over REMIND iterations? If yes, it seems that the Magpie baseline is updated each iteration with the result that total fertilizer emissions become independent of the biomass use - which to me seems wrong. 

Shouldn't only baseline biomass emissions be subtracted from the Magpie emissions, which should be fixed and not be adjusted over iterations?
so that in a standalone REMIND run with lots of biomass use, total fertilizer emissions actually do go up. 
(But maybe I misunderstand these lines)

Reply:
0. Yes, standalone only. 
2. It is in preloop, so not within the loop.
3. `pm_pebiolc_demandmag` is an exogenous parameter that is *not* dependent on model behaviour. According to the current REMIND scenario is selected from a lookup table providing different biomass demands for different SSPs and RPCs.

Reply:
thanks for the explanation!